techempower/scripts/perf/run-tfb-qrh-openshift.sh

```diff
@@ -0,0 +1,400 @@
+	echo "CMD = ${cmd}" >> setup.log
+	#sleep 30
+	${cmd} > ${RESULTS_LOG}
+	sleep 1
+}
+
+# Run the wrk load on each instace of the application
+# input: Result directory, Type of run(warmup|measure), iteration number
+# output: call the run_wrk_workload for each application service
+function run_wrk_with_scaling()
+{	
+	RESULTS_DIR_J=$1
+	TYPE=$2
+	RUN=$3
+	svc_apis=($(oc status --namespace=${NAMESPACE} | grep "tfb-qrh" | grep port | cut -d " " -f1 | cut -d "/" -f3))
+	load_setup
+	for svc_api  in "${svc_apis[@]}"
+	do
+		RESULT_LOG=${RESULTS_DIR_J}/wrk-${svc_api}-${TYPE}-${RUN}.log
+		run_wrk_workload ${svc_api} ${RESULT_LOG} &
+	done
+}
+
+# Perform warmup and measure runs
+# input: number of runs(warmup|measure), result directory 
+# output: Cpu info, memory info, node info, wrk load for each runs(warmup|measure) in the form of jason files
+function runItr()
+{
+	TYPE=$1
+	RUNS=$2
+	RESULTS_runItr=$3
+	for (( run=0; run<${RUNS}; run++ ))
+	do
+		# Check if the application is running
+		check_app 
+		echo "##### ${TYPE} ${run}" >> setup.log
+		# Get CPU and MEM info through prometheus queries
+		${SCRIPT_REPO}/perf/getmetrics-promql.sh ${TYPE}-${run} ${CPU_MEM_DURATION} ${RESULTS_runItr} ${BENCHMARK_SERVER} tfb-qrh &
+		# Run the wrk workload
+		run_wrk_with_scaling ${RESULTS_runItr} ${TYPE} ${run}
+		# Sleep till the wrk load completes
+		sleep ${DURATION}
+		sleep 1
+	done
+}
+
+# get the kruize recommendation for tfb application
+# input: result directory
+# output: kruize recommendations for tfb
+function get_recommendations_from_kruize()
+{
+	TOKEN=`oc whoami --show-token`
+	app_list=($(oc get deployments --namespace=${NAMESPACE} | grep "tfb-qrh" | cut -d " " -f1))
+	for app in "${app_list[@]}"
+	do
+		curl --silent -k -H "Authorization: Bearer ${TOKEN}" http://kruize-openshift-monitoring.apps.${BENCHMARK_SERVER}/recommendations?application_name=${app} > ${RESULTS_DIR_I}/${app}-recommendations.log
+		err_exit "Error: could not generate the recommendations for tfb-qrh" >> setup.log
+	done
+}
+
+# Perform warmup and measure runs
+# input: scaling instance, total number of iterations, warmups , measures , result directory
+# output: Deploy the application if required, perform the runs and get the recommendations
+function runIterations() {
+	SCALING=$1
+	TOTAL_ITR=$2
+	WARMUPS=$3
+	MEASURES=$4
+	RESULTS_DIR_ITR=$5
+	#IF we want to use array of users we can use this variable.
+	USERS=${WRK_LOAD_USERS}
+	for (( itr=0; itr<${TOTAL_ITR}; itr++ ))
+	do
+		if [ ${RE_DEPLOY} == "true" ]; then
+			${SCRIPT_REPO}/tfb-qrh-deploy-openshift.sh -s ${BENCHMARK_SERVER} -i ${SCALING} -g ${TFB_IMAGE} --cpureq=${CPU_REQ} --memreq=${MEM_REQ} --cpulim=${CPU_LIM} --memlim=${MEM_LIM} --maxinlinelevel=${maxinlinelevel} --quarkustpcorethreads=${quarkustpcorethreads} --quarkustpqueuesize=${quarkustpqueuesize} --quarkusdatasourcejdbcminsize=${quarkusdatasourcejdbcminsize} --quarkusdatasourcejdbcmaxsize=${quarkusdatasourcejdbcmaxsize} >> setup.log 
```

Comment:
setup.log should be created in results dir

Reply:
Also had few echo statements to display progress. Below is a run, where nothing was displayed when I ran the script, and it just logs some metrics on failure.

```
./run-tfb-qrh-openshift.sh -s <Benchmark server> -e /home/quicklab/TFB-results -u 500 -d 600 -w 0 -m 1 -r

SCALE ,  THROUGHPUT_RATE_3m , RESPONSE_TIME_RATE_3m , THROUGHPUT , RESPONSE_TIME , MAX_RESPONSE_TIME , RESPONSE_TIME_50p , RESPONSE_TIME_95p , RESPONSE_TIME_98p , RESPONSE_TIME_99p , RESPONSE_TIME_999p , MEM_USAGE , CPU_USAGE , CPU_MIN , CPU_MAX , MEM_MIN , MEM_MAX , thrpt_prom_ci , rsp_prom_ci	, Throughput , Responsetime , RESPONSETIME_MAX , STDEV_RESPTIME_MAX , WEB_ERRORS , thrp_wrk_ci , rsp_wrk_ci	, CPU_REQ , MEM_REQ , CPU_LIM , MEM_LIM , maxinlinelevel , quarkustpcorethreads , quarkustpqueuesize , quarkusdatasourcejdbcminsize , quarkusdatasourcejdbcmaxsize
1 , 99999 , 99999 , 99999 , 99999 , 99999 , 999999 , 99999 , 99999 , 99999 , 99999 , 99999 , 99999 , 99999 , 99999 , 99999 , 99999 , 0 , 0	, 99999 , 99999 , 99999 , 99999 , 9999 , 0 , 0	,  ,  ,  ,  ,  ,  ,  ,  , 
```